### DISCUSSION TITLE
Question: Keypad input

### DISCUSSION THREAD

**Discussion by dmachtey** [taligentx/dscKeybusInterface#70] [dscKeybusInterface Discussion]
I'm using dscKeybusinterface with a PC585 with HomeAssistant-MQTT example and nodemcu esp8266, and it works great with no sw modifications. Thank you for the great job you have done.

I'm wondering, now, if there is any way to get the keypad input status?

thank you 
Damian

**Comment by slipx06** [comment]
Just to add this works perfectly on my PC5005 using the HomeAssistant-MQTT example and nodemcu. I'm aslo missing the keypad but this is because the entity reports the code_format as "null" instead of a number. 

![image](https://user-images.githubusercontent.com/7227275/55670804-02aa7480-5889-11e9-8343-b252829edd71.png)

**Comment by taligentx** [comment]
Hi @dmachtey - great to hear it's working with the PC585, added to the verified panels list.  I'm not clear on what you mean about the keypad input status, which info are you looking for?

@slipx06 - same, thanks for the info on the PC5005.  The 'code' option in the [Home Assistant MQTT Alarm Control Panel component ](https://www.home-assistant.io/components/alarm_control_panel.mqtt/) is to enter a code to arm/disarm the system if necessary.  In this case, it isn't necessary - the code is entered directly in the sketch and this interface will handle entering the code as needed.

**Comment by UltimatPronin** [comment]
@taligentx, maybe @dmachtey mean something same as me in #86

**Comment by taligentx** [comment]
Closing, feel free to re-open if the question is still pertinent.

**Comment by gabter1** [comment]
I don't want to open a new issue, but I have a similar question. I am using the Home Assistant sketch without any changes on a PC5020, so you can add it to the verified list.
I would like to take out the security code from the sketch, as I believe it's a security risk, and would like to use the keypad in the alarm panel of HA. I added "code_format: number" to the configuration.yaml and now HA is showing the keypad. Can you tell me what needs to be changed in the sketch to actually read the code from the keypad and use that as the disarm code?
